napari/layers/transforms.py

```diff
@@ -41,14 +41,22 @@ def inverse(self):
             raise ValueError('Inverse function was not provided.')
 
 
-class TransformChain(Transform, ListModel):
+class TransformChain(ListModel, Transform):
     def __init__(self, transforms=[]):
-        super().__init__(basetype=Transform, iterable=transforms, lookup=id)
+        super().__init__(
+            basetype=Transform,
+            iterable=transforms,
+            lookup={str: lambda q, e: q == e.name},
+        )
 
     def __call__(self, coords):
         return tz.pipe(coords, *self)
 
+    def __newlike__(self, iterable):
+        return ListModel(self._basetype, iterable, self._lookup)
+
     def set_slice(self, axes: Sequence[int]) -> TransformChain:
+        self._set_transform_attributes()
```

Comment:
@GenevieveBuckley this line needs to be removed!

Reply:
```suggestion
```

Reply:
Good catch, thank you!